{"entity":{"name":"jediとかvirtualenvとかがReason: image not foundと言ってぶっ壊れたときの対処","nameSegmented":["jediとか","virtualenvと","かが","Reason: ","image ","not ","foundと","言って","ぶっ壊れた","ときの","対処"],"kind":"note","urlPath":"/notes/jedi-virtualenv","historyURL":"https://github.com/kissge/yo.eki.do/commits/master/notes/jedi-virtualenv.md","lastModified":"2022-06-11T08:16:13.000Z","attributes":{"from":"wordpress","title":"jediとかvirtualenvとかがReason: image not foundと言ってぶっ壊れたときの対処","date":"2016-08-26T00:17:53.000Z"},"tags":[],"body":"<p>ある日突然jedi（EmacsでPythonを書くときにつかうすごいやつだよ）がぶっ壊れた。\npython-modeのバッファでピリオドを打つたびに</p>\n<pre><code>Error (jedi): Failed to start Jedi EPC server.\n*** You may need to run &quot;M-x jedi:install-server&quot;. ***\nThis could solve the problem especially if you haven&#39;t run the command yet\nsince Jedi.el installation or update and if the server complains about\nPython module imports.\n</code></pre>\n<p>というメッセージが出てくるし、当然補完機能は使えない。\n指示通りコマンドを実行しても何も起きない。</p>\n<!--more-->\n\n<p>よくみると*Messages*に</p>\n<pre><code>Server may raise an error. Use &quot;M-x epc:pop-to-last-server-process-buffer RET&quot; to see full traceback:\n</code></pre>\n<p>とあるのでtracebackを見てみると</p>\n<pre><code>dyld: Library not loaded: *****\n  Referenced from: *****:\n  Reason: image not found\n</code></pre>\n<p>という見慣れないエラーがある。なんだこれ。</p>\n<h3 id=\"原因\">原因</h3>\n<p>簡単に言うとvirtualenvが壊れていた。\n私の環境では <code>(custom-set-variables &#39;(jedi:environment-root &quot;jedi&quot;))</code> という<a href=\"https://github.com/kissge/.emacs.d/blob/master/inits/15-python.el#L8\">感じ</a>でvirtualenvを使う設定になっているのだが、そのディレクトリの中にある大量のシンボリックリンクがいずれもリンク切れになっていた。\nhomebrewでPythonをバージョンアップしたためである。</p>\n<p>と、いうわけで、このディレクトリを削除して作りなおすと万事うまくいった。よかった。</p>\n<p>あんまし関係無いけど、この問題に対処する隙がない間エラーを無視してとりあえずjediを <code>el-get-remove</code> しようとしたら <code>auto-complete</code> がないから無理〜と言って怒られたのは納得がいかない。\njediさん何とかして。</p>\n","headline":"ある日突然jedi（EmacsでPythonを書くときにつかうすごいやつだよ）がぶっ壊れた。python-modeのバッファでピリオドを打つたびにError (jedi): Failed to start Jedi EPC server. *** You may need to run &quot;M-x jedi:install-server&quot;. ***This could solve the problem especially if you haven&#39;t run the command yet since Jedi.el installation or update and if the server complains about Python module imports. というメッセージが出てくるし、当然補完機能は使えない。指示通りコマンドを実行しても何も起きない。よくみると*Messages*にServer may raise an error. Use &quot;M-x epc:pop-to-last-server-process-buffer RET&quot; to","links":{"to":{"urlPath":"/notes/jedi-virtualenv","entities":[]},"from":{"urlPath":"/notes/jedi-virtualenv","entities":[]},"kind":{"urlPath":"/notes/jedi-virtualenv","entities":[]}}}}